dotnet/test/AutoGen.OpenAI.Tests/OpenAIChatAgentTest.cs

```diff
@@ -236,4 +228,52 @@ await foreach (var streamingMessage in reply)
             }
         }
     }
+
+    [ApiKeyFact("AZURE_OPENAI_API_KEY", "AZURE_OPENAI_ENDPOINT", "AZURE_OPENAI_DEPLOY_NAME")]
+    public async Task ItCreateOpenAIChatAgentWithChatCompletionOptionAsync()
+    {
+        var deployName = Environment.GetEnvironmentVariable("AZURE_OPENAI_DEPLOY_NAME") ?? throw new Exception("Please set AZURE_OPENAI_DEPLOY_NAME environment variable.");
+        var openaiClient = CreateOpenAIClientFromAzureOpenAI(deployName);
+        var options = new ChatCompletionsOptions(deployName, [])
+        {
+            Temperature = 0.7f,
+            MaxTokens = 1,
+        };
+
+        var openAIChatAgent = new OpenAIChatAgent(
+            openAIClient: openaiClient,
+            name: "assistant",
+            options: options)
+            .RegisterMessageConnector();
+
+        var respond = await openAIChatAgent.SendAsync("hello");
+        respond.GetContent()?.Should().NotBeNullOrEmpty();
+    }
+
+    [ApiKeyFact("AZURE_OPENAI_API_KEY", "AZURE_OPENAI_ENDPOINT", "AZURE_OPENAI_DEPLOY_NAME")]
+    public async Task ItThrowExceptionWhenChatCompletionOptionContainsMessages()
+    {
+        var deployName = Environment.GetEnvironmentVariable("AZURE_OPENAI_DEPLOY_NAME") ?? throw new Exception("Please set AZURE_OPENAI_DEPLOY_NAME environment variable.");
+        var openaiClient = CreateOpenAIClientFromAzureOpenAI(deployName);
+        var options = new ChatCompletionsOptions(deployName, [new ChatRequestUserMessage("hi")])
+        {
+            Temperature = 0.7f,
+            MaxTokens = 1,
+        };
+
+        var action = () => new OpenAIChatAgent(
+            openAIClient: openaiClient,
+            name: "assistant",
+            options: options)
+            .RegisterMessageConnector();
+
+        action.Should().ThrowExactly<ArgumentException>().WithMessage("Messages should not be provided in options");
+    }
+
+    private OpenAIClient CreateOpenAIClientFromAzureOpenAI(string deployName)
```

Comment:
Doesn't seem like `deployName` is being used here

Reply:
Because it creates `OpenAIClient` but the `deployName` is used in `ChatCompletionOption`.

Reply:
yes i agree that `deployName` is used for `ChatcompletionOption`, but should we remove `deployName` as a parameter since `CreateOpenAIClientFromAzureOpenAI` does not use it in the implementation ?

Reply:
OH, I see, sure.